src/result/result.mli

```diff
@@ -13,8 +13,18 @@ include sig
     | Error of 'error
 end
 
-(** Map successful results *)
-val map : ('a, 'error) t -> f:('a -> 'b) -> ('b, 'error) t
+val is_ok    : _ t -> bool
+val is_error : _ t -> bool
+
+module O : sig
+  val ( >>| ) : ('a, 'error) t -> ('a -> 'b) -> ('b, 'error) t
+  val ( >>= ) : ('a, 'error) t -> ('a -> ('b, 'error) t) -> ('b, 'error) t
+end
+
+val map  : ('a, 'error) t -> f:('a -> 'b) -> ('b, 'error) t
+val bind : ('a, 'error) t -> f:('a -> ('b, 'error) t) -> ('b, 'error) t
+
+val map_error : ('a, 'error1) t -> f:('error1 -> 'error2) -> ('a, 'error2) t
```

Comment:
Rather than extending this module, we should consider just vendoring rresult.ml I think.

Reply:
I think we should limit the vendoring to things to things that provide complex features or that are dependencies of things we vendor. `Result` is a very simple module. Moreover the API conventions of rresult are quite different from the rest of the Dune codebase